commit e9c90117372dc06914cf215062b7a183e8e07402
Author: Mark Robert Miller <markrmiller@apache.org>
Date:   Sat Mar 3 15:48:34 2012 +0000

    add a long leader wait at the start of this test for slow/busy comps
    
    git-svn-id: https://svn.apache.org/repos/asf/lucene/dev/trunk@1296646 13f79535-47bb-0310-9956-ffa450edef68

diff --git a/solr/core/src/test/org/apache/solr/cloud/BasicZkTest.java b/solr/core/src/test/org/apache/solr/cloud/BasicZkTest.java
index 694cf6b..bb5378d 100644
--- a/solr/core/src/test/org/apache/solr/cloud/BasicZkTest.java
+++ b/solr/core/src/test/org/apache/solr/cloud/BasicZkTest.java
@@ -46,9 +46,13 @@ public class BasicZkTest extends AbstractZkTestCase {
   
   @Test
   public void testBasic() throws Exception {
+    
     // test using ZooKeeper
     assertTrue("Not using ZooKeeper", h.getCoreContainer().isZooKeeperAware());
     
+    // for the really slow/busy computer, we wait to make sure we have a leader before starting
+    h.getCoreContainer().getZkController().getZkStateReader().getLeaderUrl("collection1", "shard1", 30000);
+    
     ZkController zkController = h.getCoreContainer().getZkController();
     
     // test merge factor picked up
@@ -154,6 +158,7 @@ public class BasicZkTest extends AbstractZkTestCase {
       
     }
     
+    zkController.getZkClient().printLayoutToStdOut();
   }
   
   public SolrQueryRequest request(String... q) {
